### ISSUE TITLE
Builtin pagination template doesn´t work with sub root in BaseURL and CanonifyUrls = "false"

### ISSUE THREAD

**Issue by donlzx** [spf13/hugo#1252]
If .Site.BaseURL contain paths (e.g., example.com/en/), pagination doesn't work correctly.  The variables URL, Next.URL, Prev.URL, etc. return incorrect values in the pagination templates, resulting in the following behavior:

http://example.com/en/products/page/2/ (correct link) => http://example.com/products/page/2/ (links in generated pages)

I make it working by hacking the source code (hugolib/pagination.go) around line 276:

> urlFactory := newPaginationURLFactory(section)

modified as: 

> urlFactory := newPaginationURLFactory(helpers.RelURL(section))

Since I am new to Hugo and haven't looked at the source codes before, can someone check this issue and fix it in a correct way?

**Comment by bep** [comment]
I'm pretty sure the Next.URL etc. are relative to `context root`, so the examples you give cannot be the string value returned. Can you check?

**Comment by donlzx** [comment]
@bep Actually, I mean the final URL in the generated pages.

**Comment by bep** [comment]
And how do you construct the "full URL"?

**Comment by donlzx** [comment]
For testing, I'm just using the default internal template and nothing special:

```
  {{ range (.Paginator 20).Pages }}
       ....
  {{ end }}
  {{ template "_internal/pagination.html" . }}
```
The full URLs are constructed by the browser.  To be exact, the links to the pages in generated HTML looks like:

```
 <a href="/products/page/2/">2</a>
```
It should be:
```
<a href="/en/products/page/2/">2</a>
```

**Comment by bep** [comment]
OK, now I understand it.

The current workaround is to either set `CanonifyUrls = "true"` or use a custom paginator template.

`helpers.RelURL` would be the right medicine, but I would do it with the `rel` variable here:

https://github.com/spf13/hugo/blob/master/hugolib/pagination.go#L367

**Comment by donlzx** [comment]
@bep Using ```helpers.RelURL()``` with ```rel``` variable doesn't work for me, because ```rel``` variable is already of the form ```/section/page/... ``` and helper.RelURL()``` doesn't change it.

**Comment by bep** [comment]
Hmm, that sound a little bit strange. I guess this should be a safe bet, too:

https://github.com/spf13/hugo/blob/master/hugolib/pagination.go#L273

**Comment by bep** [comment]
**Note/Update:** This issue is marked as stale, and I may have said something earlier about "opening a thread on the discussion forum". Please don't.

If this is a **bug** and you can still reproduce this error on the latest <code>release</code> or the <code>master</code> branch, please reply with all of the information you have about it in order to keep the issue open.

If this is a **feature request**, and you feel that it is still relevant and valuable, please tell us why.

**Comment by aureooms** [comment]
I still have the same issue in 0.20-DEV with the default paginator. `baseurl` is `https://example.com/bla/` but the paginator generate `/page/x` urls. Is it supposed to have been fixed?

**Comment by moorereason** [comment]
Yes, it's fixed been since the 0.20 release.  Update your Hugo version.  The latest is 0.25.1.

**Comment by aureooms** [comment]
Sorry! I thought I had the latest version...
